### PR TITLE
Assert caller permission

### PR DESCRIPTION
**Pull request by cyberliem**

### CODE REVIEW

**Comment by cyberliem** [comment]
[Assert caller permissions before executing write operations](https://trello.com/c/fVzlG6g3/22-assert-caller-permissions-before-executing-write-operations)

File: src/permission_assert.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+export const assertOperator = async (contract, address) => {
+  const operators = await contract.getOperators()
+  for (var idx in operators) {
+    if (operators[idx].toLowerCase() === address.toLowerCase()) {
```
Comment (favadi) [inline src/permission_assert.js]: we need 'hasOwnProperty` check here. Detail explanation: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty.

File: test/reserve_contract.test.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ import {
 const provider = ganache.provider()
 const web3 = new Web3(provider)
 
-const assertThrowAsync = async fn => {
+export const assertThrowAsync = async fn => {
```
Comment (favadi) [inline test/reserve_contract.test.js]: please move this to a a common package, it is not belong to reserve_contract.

File: src/permission_assert.js (hunk 1 above)
Comment (favadi) [reply src/permission_assert.js]: I created https://github.com/KyberNetwork/reserve-sdk.js/pull/27 for this.

**Pull request merged**